x/ecocredit/client/testsuite/grpc.go

```diff
@@ -214,64 +164,47 @@ func (s *IntegrationTestSuite) TestGetBatch() {
 			resp, err := rest.GetRequest(tc.url)
 			require.NoError(err)
 
-			var batch ecocredit.QueryBatchInfoResponse
+			var batch core.QueryBatchInfoResponse
 			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &batch)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
-			} else {
-				require.NoError(err)
-				require.NotNil(batch.Info)
-				require.Equal(batch.Info.ProjectId, tc.projectID)
-			}
+			require.NoError(err)
+			require.NotNil(batch.Info)
+			require.Equal(batch.Info.BatchDenom, batchDenom)
 		})
 	}
 }
 
 func (s *IntegrationTestSuite) TestCreditTypes() {
 	require := s.Require()
 	val := s.network.Validators[0]
+	creditTypes := core.DefaultParams().CreditTypes
 
-	url := fmt.Sprintf("%s/regen/ecocredit/v1alpha2/credit-types", val.APIAddress)
+	url := fmt.Sprintf("%s%scredit-types", val.APIAddress, coreRoute)
 	resp, err := rest.GetRequest(url)
 	require.NoError(err)
 
-	var creditTypes ecocredit.QueryCreditTypesResponse
-	err = val.ClientCtx.Codec.UnmarshalJSON(resp, &creditTypes)
-
+	var res core.QueryCreditTypesResponse
+	err = val.ClientCtx.Codec.UnmarshalJSON(resp, &res)
 	require.NoError(err)
-	require.Len(creditTypes.CreditTypes, 1)
-	require.Equal(creditTypes.CreditTypes[0].Abbreviation, "C")
-	require.Equal(creditTypes.CreditTypes[0].Name, "carbon")
+	require.Equal(creditTypes, res.CreditTypes)
+
 }
 
 func (s *IntegrationTestSuite) TestGetBalance() {
 	val := s.network.Validators[0]
+	noBalAddr := s.network.Validators[1]
+	_, _, batchDenom := s.createClassProjectBatch(val.ClientCtx, val.Address.String())
 
 	testCases := []struct {
-		name   string
-		url    string
-		expErr bool
-		errMsg string
+		name string
+		url  string
 	}{
 		{
-			"invalid batch-denom",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha2/batches/%s/balance/%s", val.APIAddress, "abcd", val.Address.String()),
-			true,
-			"invalid denom",
+			"valid request",
+			fmt.Sprintf("%s%sbatches/%s/balance/%s", val.APIAddress, coreRoute, batchDenom, val.Address.String()),
 		},
 		{
-			"invalid account address",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha2/batches/%s/balance/%s", val.APIAddress, "C01-20210101-20210201-001", "abcd"),
-			true,
-			"decoding bech32 failed",
-		},
```

Comment:
I think these are valuable test cases but maybe not necessary if redundant of the unit tests...?

Reply:
yeah these seem like smaller, unit-level test cases imo. I don't think our integration tests should really be concerned with these types of invalid inputs

Reply:
We did catch an issue with create batch so we should probably do an audit of our current unit tests. I think this is the right direction though. I'm feeling a bit caught between further revising our unit tests and writing acceptance tests that would potentially replace those. This seems safe since these cases should be covered in message validation.